VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -975,15 +966,19 @@ private boolean isScreenLandscape() {
         @Override
         public void run() {
             if (isResumed() && !isRemoving()) {
-                hideProgress();
-                if(player!=null) {
+                if (player != null) {
                     player.unfreeze();
+                    hideProgress();
                     if (player.isPlaying()) {
-                        hideProgress();
                         updateController("player unfreezed");
                     }
 
-                    if (pauseDueToDialog){
+                    if (IS_AUTOPLAY_ENABLED && isPrepared && !isAutoPlayDone) {
+                        isAutoPlayDone = true;
+                        player.start();
```

Comment:
Why not? There is a check around the start of the `onPrepared()` implementation, which returns from the method if `getUserVisibleHint()` returns false. If anything, the scenario isn't being handled in the current implementation.